dashboard/app/helpers/levels_helper.rb

```diff
@@ -588,13 +588,14 @@ def redirect_under_13_without_tos_teacher(level)
     return false unless level.game == Game.applab || level.game == Game.gamelab
 
     if current_user && current_user.under_13? && current_user.terms_version.nil?
-      redirect_to '/', :flash => { :alert => I18n.t("errors.messages.too_young") }
+      error_message = current_user.student_of_any_teacher? ? I18n.t("errors.messages.teacher_must_accept_terms") : I18n.t("errors.messages.too_young")
```

Comment:
Honestly I think `student.teachers.any?` is far more clear than `student_of_any_teacher?` (also less symbols). My vote is to eliminate the helper method, though feel welcome to disagree.